src/display-x11.cc

```diff
@@ -945,18 +945,14 @@ void display_output_x11::move_win(int x, int y) {
 #endif /* OWN_WINDOW */
 }
 
-const size_t PIXELS_PER_INCH = 96;
-template <typename T, typename>
-T display_output_x11::dpi_scale(T value) {
-#if defined(BUILD_XFT)
+const float PIXELS_PER_INCH = 96.0;
+float display_output_x11::get_dpi_scale() {
+#ifdef BUILD_XFT
   if (use_xft.get(*state) && xft_dpi > 0) {
-    return (value * xft_dpi + (value > 0 ? 48 : -48)) / PIXELS_PER_INCH;
```

Comment:
I believe `(value > 0 ? 48 : -48)` was rounding the result for `int` values before, so I replaced it with `ceil` and `floor` in new code.